### ISSUE TITLE
tap-next-release and order of release events

### ISSUE THREAD

**Issue by jchtt** [kmonad/kmonad#256]
I mentioned this particular problem before in #111, see [here](https://github.com/kmonad/kmonad/issues/111#issuecomment-716747605). When using `tap-next-release` (and `tap-hold-next-release`), key release events of other keys are emitted immediately, while I think it would be preferable to put them on hold until the `tap-next-release` key is resolved.

For example, consider this configuration:
```
(defsrc a)
(deflayer default @act)
(defalias
  act (tap-next-release b lctl)
)
```
When inputting `Pshift Pa Rshift Ra`, kmonad currently translates this into `Pshift Rshift Pb Rb` (a lowercase `b`). Since I have some `tap-hold-next-release` buttons on my home row, this behavior often leads to typos when I want to type the associated letters in uppercase since I am not holding the shift key long enough to finish the release of the other key.

I think it would be better if the above got translated into `Pshift Pb Rshift Rb` (an uppercase `B`), keeping the original order of key presses and releases intact. This also seems overall more predictable for the user.

**Comment by thoelze1** [comment]
I agree that `tap-next-release` (and `tap-hold-next-release`) should behave as you describe.

 I couldn't figure out how to fix this on `master`. The `tap-next-release` button in KMonad.Model.Button pauses processing of other events with `hold True` and later unpauses it with `hold False`. It appears that `hold` only pauses processing of as-yet-unpressed buttons, and so it does not (cannot?) pause the release of `shift` (because it's already been pressed). Maybe there is a simple fix on `master` that I don't know how to implement, or maybe this will be easier to change in `keycode-refactor`. Any insight @david-janssen?

**Comment by david-janssen** [comment]
> Maybe there is a simple fix on master that I don't know how to implement, or maybe this will be easier to change in keycode-refactor. 

Sadly neither, however, that is why I am already dreaming of `model-refactor` :)
